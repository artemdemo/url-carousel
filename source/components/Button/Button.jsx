import React from 'react';
import classnames from 'classnames';

const Button = (props) => {
    const { className, primary, onClick } = props;
    const buttonClass = classnames(className, {
        btn: true,
        'btn-primary': primary,
    });

    return (
        <button className={buttonClass} onClick={e => onClick && onClick(e)}>
            {props.children}
        </button>
    );
};

Button.propTypes = {
    className: React.PropTypes.string,
    onClick: React.PropTypes.func,
    primary: React.PropTypes.bool,
};

export default Button;
