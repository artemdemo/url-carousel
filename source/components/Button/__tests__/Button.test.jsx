import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';

describe('<Button>', () => {
    it('Simple render with text', () => {
        const tree = renderer.create(
            <Button>
                button name
            </Button>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('With className', () => {
        const tree = renderer.create(
            <Button className='additional-class'>
                button name
            </Button>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('With className and "primary"', () => {
        const tree = renderer.create(
            <Button className='additional-class' primary>
                button name
            </Button>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
