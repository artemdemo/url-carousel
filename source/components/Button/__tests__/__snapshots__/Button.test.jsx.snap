// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> Simple render with text 1`] = `
<button
  className="btn"
  onClick={[Function]}
>
  button name
</button>
`;

exports[`<Button> With className 1`] = `
<button
  className="additional-class btn"
  onClick={[Function]}
>
  button name
</button>
`;

exports[`<Button> With className and "primary" 1`] = `
<button
  className="additional-class btn btn-primary"
  onClick={[Function]}
>
  button name
</button>
`;
